### PR TITLE
NECO reply and changes

### PR DESCRIPTION
**Pull request by arvoelke**

### CODE REVIEW

**Comment by arvoelke** [comment]
Resubmit TODO:
 - [x] Update README
 - [x] Address issue #1 
 - [x] Check Andreas' report

Camera-ready TODO:
 - [ ] Include DOI for Voelker & Eliasmith 2017 citation (http://arxiv.org/abs/1708.08133)
 - [ ] Release nengolib==0.4.1

File: delay2017.tex [hunk 1]
```diff
@@ -154,16 +154,28 @@ \section{Introduction}
...
+We provide a novel derivation of an optimal low-dimensional linear approximation to a continuous-time delay, and prove that the resulting \emph{delay network} nonlinearly encodes its input across the delay interval.
+This network is scale-invariant, with a level of accuracy that depends on the input frequency, chosen dimensionality (i.e.,~the order of the approximation), and particular synapse model.
 Low-dimensional representations (e.g.,~$\le 27$) of low-frequency signals (e.g.,~$\le 50$\,Hz) are pervasive in biological systems~\citep{cunningham2014dimensionality, waernberg2017low, pulvermuller1997high, singer1999neuronal}.
 To our knowledge, this work is the first to demonstrate that such a temporal code may be accurately implemented using a spiking dynamical network.
 
+Reservoir Computing approaches, such as Liquid State Machines~\citep{maass2002real} and Echo State Networks~\citep{jaeger2001echo}, may be used to approximate a delay line.
+However, since these networks use randomly chosen feedback weights, there is no good reason to believe they {\it efficiently} produce the dynamics of a pure delay.
```
Comment (celiasmith) [inline delay2017.tex]: - "there is no good reason to believe they" -> it is likely that they do not

File: delay2017.tex [hunk 2]
```diff
@@ -154,16 +154,28 @@ \section{Introduction}
...
+This network uses a scale-invariant representation, with a level of accuracy that depends on the input frequency, chosen dimensionality (i.e.,~the order of the approximation), and particular synapse model.
 Low-dimensional representations (e.g.,~$\le 27$) of low-frequency signals (e.g.,~$\le 50$\,Hz) are pervasive in biological systems~\citep{cunningham2014dimensionality, waernberg2017low, pulvermuller1997high, singer1999neuronal}.
 To our knowledge, this work is the first to demonstrate that such a temporal code may be accurately implemented using a spiking dynamical network.
 
+Reservoir Computing approaches, such as Liquid State Machines~\citep{maass2002real} and Echo State Networks~\citep{jaeger2001echo}, may be used to approximate a delay line.
+However, since these networks use randomly chosen feedback weights, we have no good reason to believe they {\it efficiently} produce the dynamics of a pure delay.
+Rather, these networks are understood to represent a random variety of nonlinear memory traces~\citep{lukovsevicius2012reservoir}.
```
Comment (celiasmith) [inline delay2017.tex]: delete " are understood to "

File: delay2017.tex [hunk 3]
```diff
@@ -154,16 +154,28 @@ \section{Introduction}
...
+Discrete approaches to short-term memory, such as those taken by \citet{white2004short} and \citet{ganguli2008memory}, while optimal in an information-theoretic sense, rely fundamentally on single time-step delays between rate-based neurons.
+In contrast, the method that we propose here works independently of the simulation time-step, and is optimal assuming the population of spiking neurons---coupled with some model of the synapse---accurately represents a low-dimensional, low-frequency, vector space.
+Furthermore, our framework is extended to account for arbitrary linear synapses, which improves our understanding of the relationship between synapse models and network-level computation.
+A detailed comparison of our method to Reservoir Computing remains the subject of future work.
+
+A distinguishing feature of this work is that we begin the problem with a model description of the ideal dynamics for the delay line, and then proceed by mapping this model onto a spiking neural network (using the NEF and its extensions).
```
Comment (celiasmith) [inline delay2017.tex]: -model->mathematical
-model->description

File: reply-letter.tex [hunk 4]
```diff
@@ -97,7 +103,7 @@ \section*{Reviewer \# 1}
 
 \REPLY{1.7. Previously, this was being done (with $\tau=0.02$\,s) because we were decoding $y(t) = C{\mathbf x}(t)$ from spikes.
 Since this is a weighted combination of spikes, some filter is required.
-However, as you likely realized, we can just use the PSC from the recurrent connection to get ${\mathbf x}(t)$---without needing any additional filtering---and then simply transform this by $C$.
+However, as may be evident, we can use the PSC from the recurrent connection to get ${\mathbf x}(t)$---without needing any additional filtering---and then simply transform this by $C$.
```
Comment (celiasmith) [inline reply-letter.tex]: remove 'simply'

File: reply-letter.tex [hunk 5]
```diff
@@ -146,68 +161,71 @@ \section*{Reviewer \# 1}
...
 
-\REPLY{2.1. The gamma model is related to our work in several ways, but is also quite distinct in some crucial ways that we will discuss now. 
-First, spiking neurons are not considered at all in the paper by Bert de Vries and Jose C. Principe or in any of the literature we found in section~2.
-Second, their use of the term ``delay kernel'' is referring to the fact that the output of a temporal convolution at time $t$ depends on the input at time $t - s$ (e.g.,~equation~8), as specified by some kernel of weights.
-In particular, they refer to equation~29 as a ``delay structure'' despite it being a set of differential equations implementing their gamma kernel.
-This is more appropriately referred to in the signal processing literature as a particular kind of lowpass filter (more on this below).
+\REPLY{2.1. First, spiking neurons are not considered in the paper by Bert de Vries and Jose C. Principe or in the literature we found in section~2 of VP.
```
Comment (celiasmith) [inline reply-letter.tex]: We agree that the gamma model is closely related and have significantly extended the introduction to incorporate a discussion highlighting similarities and differences.  We begin by highlighting some of the important differences.

File: reply-letter.tex [hunk 6]
```diff
@@ -146,68 +161,71 @@ \section*{Reviewer \# 1}
...
+Second, their focus is on using backpropagation to train the weights to minimize some supervised error signal, while our methods are focused on analytically mapping prescribed model dynamics onto arbitrary linear synapses.
+How backpropagation learning may or may not be able to learn such functions in networks of spiking neurons (which have discontinuous gradients) is a discussion beyond our scope, but remains a field of active research (e.g.,~``Gradient Descent for Spiking Neural Networks'' by Sejnowski~(2017) and ``Deep learning in spiking LIF neurons'' by Hunsberger~(2017)).
```
Comment (celiasmith) [inline reply-letter.tex]: Huh & Sejnowski (2017)
Hunsberger & Eliasmith (2017)

File: reply-letter.tex (hunk 5 above)
Comment (celiasmith) [inline reply-letter.tex]: define VP (i.e. after the name say 'which we will refer to as VP')

**Pull request merged**